app/tests/src/wallet/account-manager.js

```diff
@@ -45,62 +40,40 @@ const initAccountManager = async (settings, i) => {
     config,
   })
 
-  accountManagers.push(
-    AccountManager({config, cryptoProvider, blockchainExplorer, maxAccountIndex})
-  )
+  return AccountManager({config, cryptoProvider, blockchainExplorer, maxAccountIndex})
 }
 
-before(async () => {
-  for (const accountManagerSetting of accountManagerSettings) {
-    await initAccountManager(accountManagerSetting)
-  }
+before(() => {
+  Object.entries(accountManagerSettings).forEach(([name, setting]) => {
```

Comment:
`initAccountManager` seems to be async so this can lead to flaky tests if I understand correctly. `await Promise.all(Object.entries(accountManagerSettings).map(etc))` should address that

Reply:
Yes, the `initAccountManager` is still async, but I decided not to use `Promise.all()` as you suggest, since it returns an array and also expects an array of promises, and that would forfeit the whole idea behind having the settings as an object. Therefore I decided to store the AccountManagers as Promises. Anyway I could refactor it to for cycle to keep the object behaviour and store awaited values instead of promises. What do you think?

Reply:
Promise.all awaits on an array of promises, the returned result is just a byproduct which we don't really care about here as you are modifying an object from the outer scope anyway, but a for loop with awaits is also an option of course and arguably more readable though the linter may complain if this rule is enabled which may be as in production this usually results in suboptimal performace given that the promises are chained instead of executed in parallel which doesn't really matter here: https://eslint.org/docs/rules/no-await-in-loop

Reply:
Okay, `Promise.all` it is :+1: Thnx